float64[] s_des

